### PR TITLE
Upgrade use of generic ptr to cleaner codebase.

### PR DESCRIPTION
**Pull request by zzjin**
**What type of PR is this?**

chore: change pointer*ptr usage to cleaner codebase.

**What this PR does / why we need it**:

1. Upgrade usage from deprecated `k8s.io/utils/pointer` to generic version `k8s.io/utils/ptr`
2. Remove `internal/utils/ptr` cause as duplicated
3. Simpler some `ptr.To` call

### CODE REVIEW

**Review by Xunzhuo — DISMISSED** [review]
lgtm

File: api/v1alpha1/kubernetes_helpers.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	appv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	"k8s.io/utils/pointer"
```
Comment (zirain) [inline api/v1alpha1/kubernetes_helpers.go:12]: can you also update golanglint-ci setting for this package?

Comment (zirain) [reply api/v1alpha1/kubernetes_helpers.go:12]: https://github.com/envoyproxy/gateway/blob/f4252e6685a4378a9b2e75f802507711c464dfa6/tools/linter/golangci-lint/.golangci.yml#L27

Comment (zzjin) [reply api/v1alpha1/kubernetes_helpers.go:12]: Updated.

**Review by zirain — APPROVED** [review]

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**